### PR TITLE
Add SetExcludeScheme method.

### PR DESCRIPTION
**Pull request by brunnre8**
Maps to notmuch_query_set_omit_excluded and allows to change the
exclude behaviour of a query.

Please carefully look over the C calling stuff... I don't normally write C go and may have missed something

### CODE REVIEW

**Comment by zenhack** [comment]
Looks like the failures are because I haven't updated the build matrix in a while, so it's still testing on super old versions of Go.

At a glance this looks reasonable, I'll have a closer look soonish.

Thanks for the patch!

File: query.go [hunk 1]
```diff
@@ -11,9 +11,38 @@ import "C"
...
 import "unsafe"
 
+const (
+	// notmuch_query_search_messages and notmuch_query_search_threads
+	// will return all matching messages/threads regardless of exclude status.
+	// The exclude flag will be set for any excluded message that is
+	// returned by notmuch_query_search_messages, and the thread counts
+	// for threads returned by notmuch_query_search_threads will be the
+	// number of non-excluded messages/matches.
+	QUERY_EXCLUDE_FLAG = C.NOTMUCH_EXCLUDE_FLAG
```
Comment (zenhack) [inline query.go]: Let's omit the `QUERY_*` prefix, and just call it `EXCLUDE_FLAG` (and likewise with the other constants).

File: query.go [hunk 2]
```diff
@@ -11,9 +11,38 @@ import "C"
 
 import "unsafe"
 
+const (
+	// notmuch_query_search_messages and notmuch_query_search_threads
```
Comment (zenhack) [inline query.go]: Use the names for the go wrappers, rather than the c identifiers -- so in this case Query.SearchMessages.

**Review by zenhack — CHANGES_REQUESTED** [review]
A few minor comments.

I'll fuss with the CI bitrot myself soonish; in the meantime I'll just test locally.

**Comment by zenhack** [comment]
I switched us over to gitlab ci on master; you'll need to fix the merge conflicts (shouldn't be hard).

**Comment by brunnre8** [comment]
Note that Query.SearchMessages isn't actually implemented yet.... :wink: 

I thought it was, but only Query.Threads actually is...

I'll try and patch that up in a separate PR if you don't beat me to it.
Which would you prefer? Query.SearchMessages or Query.Messages?

I think we should stay consistent with the way threads are queried

**Pull request merged**